src/core/annotation.js

```diff
@@ -1367,6 +1384,76 @@ class WidgetAnnotation extends Annotation {
     }
     return localResources || Dict.empty;
   }
+
+  _collectJS(entry, xref, list, parents) {
+    if (!entry) {
+      return;
+    }
+
+    let parent = null;
+    if (isRef(entry)) {
+      if (parents.has(entry)) {
+        // If we've already found entry then we've a cycle.
+        return;
+      }
+      parent = entry;
+      parents.put(parent);
+      entry = xref.fetch(entry);
+    }
+    if (Array.isArray(entry)) {
+      for (const element of entry) {
+        this._collectJS(element, xref, list, parents);
+      }
+    } else if (entry instanceof Dict) {
+      if (isName(entry.get("S"), "JavaScript") && entry.has("JS")) {
+        const js = entry.get("JS");
+        let code;
+        if (isStream(js)) {
+          code = bytesToString(js.getBytes());
+        } else {
+          code = stringToPDFString(js);
+        }
+        if (code) {
+          list.push(code);
+        }
+      }
+      this._collectJS(entry.getRaw("Next"), xref, list, parents);
+    }
+
+    if (parent) {
+      parents.remove(parent);
+    }
+  }
+
+  _collectActions(xref, dict) {
+    const actions = Object.create({});
```

Comment:
Why not simply `Object.create(null);` here?

Reply:
isn't the same ? (I'm not a JS expert)

Reply:
Well, they might not be identical; but why do you *need* that special format here?

Reply:
dunno ;)
I'll change it.